server/src/com/cloud/template/TemplateAdapterBase.java

```diff
@@ -293,9 +298,15 @@ public TemplateProfile prepare(GetUploadParamsForTemplateCmd cmd) throws Resourc
             zoneId = -1L;
         }
 
+        HypervisorType hypervisorType = HypervisorType.getType(cmd.getHypervisor());
+        if(hypervisorType == HypervisorType.None) {
+            throw new InvalidParameterValueException("Hypervisor Type: " + cmd.getHypervisor() + " is invalid. Supported Hypervisor types are "
+                                                         + EnumUtils.listValues(HypervisorType.values()).replace("None, ", ""));
+        }
```

Comment:
@GabrielBrascher there is lot of duplicate code between the methods. A proper cleanup has to be planned. Cant take up now.